### PR TITLE
use nextest for better test isolation

### PR DESCRIPTION
**Pull request by pompon0**
it will allow us to set panic=abort per test (because each test is executed in a separate subprocess) and set test timeouts without developing a custom test harness. Additionally I've set the default test timeout to 2min (period = 1min, killing test after 2 periods), which seems reasonable, given that all presubmit tests execute <1min as of today.

### CODE REVIEW

**Comment by pugachAG** [comment]
+1 for enabling this to set timeout for individual unit tests.

I was debugging [flaky `peer_manager::tests::routing::archival_node` test](https://near.zulipchat.com/#narrow/stream/295558-pagoda.2Fcore/topic/flaky.20eer_manager.3A.3Atests.3A.3Arouting.3A.3Aarchival_node.20test) and started preparing a PR to set timeout for `cargo test` using `RUST_TEST_TIME_UNIT` env variable along with `-Z unstable-options --ensure-time`, but this seems like a nicer solution!

**Review by mm-near — APPROVED** [review]

**Review by nagisa — APPROVED** [review]

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -48,7 +51,8 @@ steps:
   - label: "cargo test nightly not integration-tests*"
     command: |
       source ~/.cargo/env && set -eux
-      cargo test --workspace --features nightly,test_features,mock_node -p '*' --exclude 'integration-tests*'
+      cargo install cargo-nextest
+      cargo nextest run --workspace --features nightly,test_features,mock_node -p '*' --exclude 'integration-tests*'
```
Comment (nagisa) [inline .buildkite/pipeline.yml:55]: Nextest does not run doc tests so we need another stage that runs `cargo test --doc`.

**Comment by nagisa** [comment]
I would argue that we want to adjust documentation to inform developers to use `nextest` locally as a followup as well. Using different test runners can and will lead to weird hard to diagnose issues (why is it passing on CI but not locally?) eventually.

**Comment by akhi3030** [comment]
> I would argue that we want to adjust documentation to inform developers to use `nextest` locally as a followup as well. Using different test runners can and will lead to weird hard to diagnose issues (why is it passing on CI but not locally?) eventually.

+1.  Please add some documentation in https://near.github.io/nearcore/practices/testing/index.html (https://github.com/near/nearcore/tree/master/docs/practices/testing)

**Comment by pompon0** [comment]
>  (why is it passing on CI but not locally?) 

We have that already, even though we are using the same test harness in CI and locally. Updated documentation.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (pompon0) [reply .buildkite/pipeline.yml:55]: done

**Pull request merged**